feat: add theme variables

runnabro opened the pull request:
[![PR App][icn]][demo] | Fix RM-XYZ
:-------------------:|:----------:

## 🧰 Changes

Started working on the callout component… and thought I'd start with some QOL features first!

Changes:
* Add all color variants of callouts to the tests
* Add color theme toggle to the header
* Add var + theme (and their associated mixing/helpers) to the markdown project
    * This should only apply to the sandbox—not the final output
    * TODO: We can also get rid of a few markdown overrides in the main repo

![Markdown Demo-20250516-170249@2x](https://github.com/user-attachments/assets/56e5142b-99da-42a5-a725-344d5f5acd39)


## 🧬 QA & Testing

- [Broken on production][prod].
- [Working in this PR app][demo].


[demo]: https://markdown-pr-PR_NUMBER.herokuapp.com
[prod]: https://SUBDOMAIN.readme.io
[icn]: https://user-images.githubusercontent.com/886627/160426047-1bee9488-305a-4145-bb2b-09d8b757d38a.svg

runnabro committed:
add theme variables

kellyjosephprice reviewed (approved):
LGTM!

Would you expand and commit the snapshot diffs here: https://github.com/readmeio/markdown/actions/runs/15079448641?pr=1105

kellyjosephprice commented:
Oops, also wanted to ask if we should add dark mode snapshots now?

rafegoldberg commented on example/styles/vars.scss:
```diff
@@ -0,0 +1,578 @@
+@import './methods/merge-multiple';
```
```suggestion
// DO NOT EDIT: THIS FILE WAS DUPED FROM THE README MONOREPO

@import './methods/merge-multiple';
```

or something to that effect?

rafegoldberg reviewed (approved):
So all these `styles/` files are a straight rip from ReadMe proper, yeah? Should we add a comment above ‘em that says as much?

rafegoldberg commented:
Ohhh actually, just realized that all these SCSS files are contained within the `example/` dir so [this ask](https://github.com/readmeio/markdown/pull/1105#pullrequestreview-2851752248) is probably not as important as I'd thought!

kevinports committed:
refactor: convert `run` and `compile` to have synchronous execution (#1108)

[![PR App][icn]][demo] | Fix RM-XYZ
:-------------------:|:----------:

We're working on removing the need for the react app to require _both_
`@readme/mdx` and `@readme/markdown` packages in order to render the
Hub, because it mushrooms our bundle sizes.

We'd like to use `@readme/mdx` to render plain markdown down like this:
```tsx
rmdx.run(rmdx.compile(doc, { format: 'md' }))
```
but we have a number of places in the React app where we render markdown
strings in-line in react templates. Some examples are:
- [The “excerpt” on a Guides
page](https://github.com/readmeio/readme/blob/next/packages/react/src/routes/Doc/Header/index.tsx#L59)
- [A discussions
post](https://github.com/readmeio/readme/blob/next/packages/react/src/routes/Discuss/Post/Excerpt.jsx#L81-L83)

and we need the markdown to get rendered on SSR. Because the mdx `run`
and `compile` functions we use are async, these renders get skipped on
SSR and cause a layout shift client side when the react app mounts and
the markdown gets rendered.

In order to support this, we need to convert `@readme/mdx` rendering to
have synchronous execution

- Convert `lib/run.tsx` and `lib/compile.tsxt` to sync functions
- Update tests

- [Broken on production][prod].
- [Working in this PR app][demo].

[demo]: https://markdown-pr-PR_NUMBER.herokuapp.com
[prod]: https://SUBDOMAIN.readme.io
[icn]:
https://user-images.githubusercontent.com/886627/160426047-1bee9488-305a-4145-bb2b-09d8b757d38a.svg

runnabro committed:
tests and comments

runnabro committed:
update snaps

runnabro committed:
update tests

runnabro committed:
tests???
